src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -129,12 +131,20 @@ public PPLTool(Client client, String modelId, String contextPrompt, String pplMo
         } else {
             this.contextPrompt = contextPrompt;
         }
+        this.previousToolKey = previousToolKey;
     }
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
         parameters = extractFromChatParameters(parameters);
-        String indexName = parameters.get("index");
+        String indexName;
+        if (parameters.containsKey("index")) {
+            indexName = parameters.get("index");
+        } else if (!StringUtils.isBlank(this.previousToolKey) && parameters.containsKey(this.previousToolKey + ".output")) {
+            indexName = parameters.get(this.previousToolKey + ".output"); // read index name from previous key
```

Comment:
should we handle 'Not sure' here ?

Reply:
Ok. Once the index routing pr is merged I will check whether the index is equal to NOT_SURE and then throw an exception if it is.